homeassistant/components/nut/config_flow.py

```diff
@@ -85,13 +87,15 @@ def __init__(self):
         self.ups_list = None
         self.title = None
 
-    async def async_step_zeroconf(self, discovery_info):
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
         """Prepare configuration for a discovered nut device."""
         self.discovery_info = discovery_info
         await self._async_handle_discovery_without_unique_id()
         self.context["title_placeholders"] = {
-            CONF_PORT: discovery_info.get(CONF_PORT, DEFAULT_PORT),
-            CONF_HOST: discovery_info[CONF_HOST],
+            CONF_PORT: discovery_info[zeroconf.ATTR_PORT],
```

Comment:
Port is a compulsory key in ZeroconfServiceInfo, so I don't think the DEFAULT_PORT fallback is needed.

Reply:
Some services will broadcast as port `0` so maybe `discovery_info[zeroconf.ATTR_PORT] or DEFAULT_PORT`